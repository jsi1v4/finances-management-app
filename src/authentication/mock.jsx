import React from 'react';

import { AuthenticationContext } from './index';

export function AuthenticationMock({ children }) {
  return (
    <AuthenticationContext.Provider
      value={{
        isAuth: true,
        userName: 'John Doe',
        authOn: jest.fn(),
        authOff: jest.fn()
      }}
    >
      {children}
    </AuthenticationContext.Provider>
  );
}

export default AuthenticationMock;
