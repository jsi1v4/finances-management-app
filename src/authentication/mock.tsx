import React from 'react';

import { AuthenticationContext } from './index';

export function AuthenticationMock({ children }) {
  return (
    <AuthenticationContext.Provider
      value={{
        isAuth: true,
        session: null,
        authOn: jest.fn(),
        authOff: jest.fn()
      }}
    >
      {children}
    </AuthenticationContext.Provider>
  );
}

export default AuthenticationMock;
