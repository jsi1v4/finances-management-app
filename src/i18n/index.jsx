import React, { createContext, useState } from 'react';
import { useIntl } from 'react-intl';

import { INIT_LOCALE } from '../config';

export function useI18n() {
  const intl = useIntl();
  return (id, values) => {
    const message = intl.formatMessage({ id }, values);
    return <>{message}</>;
  };
}

export const I18nContext = createContext(null);

export function I18nProvider({ children }) {
  const [lang, setLanguage] = useState(INIT_LOCALE);
  const changeLang = (value) => {
    setLanguage(value);
  };

  return (
    <I18nContext.Provider
      value={{
        lang,
        changeLang
      }}
    >
      {children}
    </I18nContext.Provider>
  );
}

export default I18nProvider;
