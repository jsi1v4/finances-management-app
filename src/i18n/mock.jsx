import React from 'react';

import { I18nProvider } from './index';
import { I18nMessagesProvider } from './messages';

export function I18nMock({ children }) {
  return (
    <I18nProvider>
      <I18nMessagesProvider>{children}</I18nMessagesProvider>
    </I18nProvider>
  );
}

export default I18nMock;
