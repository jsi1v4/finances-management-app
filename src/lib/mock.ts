export class Auth {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  onAuthStateChanged(fn = () => {}) {
    return () => {};
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  async setPersistence(persist?: string) {
    return new Promise((resolve) => resolve({}));
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  async signInWithEmailAndPassword(username: string, password: string) {
    return new Promise((resolve) => resolve({}));
  }

  async signOut() {
    return new Promise((resolve) => resolve({}));
  }
}

export default Auth;
