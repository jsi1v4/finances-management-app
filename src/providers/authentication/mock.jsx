import React from 'react';

import { AuthenticationContext } from './index';

export function AuthenticationMock({ children }) {
  return (
    <AuthenticationContext.Provider
      value={{
        userName: 'John Doe',
        signOut: jest.fn(),
      }}
    >
      {children}
    </AuthenticationContext.Provider>
  );
}

export default AuthenticationMock;
