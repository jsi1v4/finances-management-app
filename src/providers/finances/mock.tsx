import React from 'react';

import { FinancesContext } from './index';
import { FinancesService } from './mock-service';

export function FinancesMock({ children }) {
  return (
    <FinancesContext.Provider value={new FinancesService({})}>
      {children}
    </FinancesContext.Provider>
  );
}

export default FinancesMock;
