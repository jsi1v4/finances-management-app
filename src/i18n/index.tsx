import React, { createContext, useContext, useState } from 'react';
import { useIntl } from 'react-intl';

import { INIT_LOCALE } from 'src/config';
import { II18nContext, Locale } from './types';

export const I18nContext = createContext<II18nContext>(null as never);

export const useI18n = () => useContext(I18nContext);

export function I18nProvider({ children }) {
  const [lang, setLanguage] = useState<Locale>(INIT_LOCALE);
  const changeLang = (value: Locale) => {
    setLanguage(value);
  };

  return (
    <I18nContext.Provider
      value={{
        lang,
        changeLang
      }}
    >
      {children}
    </I18nContext.Provider>
  );
}

export function useI18nMessage() {
  const intl = useIntl();
  return (id: string, values?: Record<string, string>) => {
    const message = intl.formatMessage({ id }, values);
    return message;
  };
}
