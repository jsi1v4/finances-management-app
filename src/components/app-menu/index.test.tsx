import React from 'react';
import { render } from '@testing-library/react';

import { I18nMock } from 'src/i18n/mock';
import { AppMenu } from './index';

describe('AppMenu component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<AppMenu />, {
      wrapper: I18nMock
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render component', async () => {
    const { getByTestId } = render(<AppMenu />, {
      wrapper: I18nMock
    });
    expect(getByTestId('menu').innerHTML).toBeDefined();
  });
});
