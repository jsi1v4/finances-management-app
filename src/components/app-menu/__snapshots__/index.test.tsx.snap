// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMenu component matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
    data-testid="menu"
    role="menu"
  >
    <li
      class="ant-menu-item"
      role="menuitem"
      style="padding-left: 24px;"
    >
      <span
        aria-label="home"
        class="anticon anticon-home"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="home"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
          />
        </svg>
      </span>
      <span>
        Home
      </span>
    </li>
    <li
      class="ant-menu-item"
      role="menuitem"
      style="padding-left: 24px;"
    >
      <span
        aria-label="table"
        class="anticon anticon-table"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="table"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 208H676V232h212v136zm0 224H676V432h212v160zM412 432h200v160H412V432zm200-64H412V232h200v136zm-476 64h212v160H136V432zm0-200h212v136H136V232zm0 424h212v136H136V656zm276 0h200v136H412V656zm476 136H676V656h212v136z"
          />
        </svg>
      </span>
      <span>
        Sheet
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
